File: AkkaDispatcherExample/src/main/scala/org/akka/essentials/dispatcher/example/BalancingDispatcher/Example1.scala
package org.akka.essentials.dispatcher.example.BalancingDispatcher
import akka.actor.ActorRef
import akka.actor.ActorSystem
import com.typesafe.config.ConfigFactory
import akka.actor.Props
import org.akka.essentials.dispatcher.MsgEchoActor
import akka.routing.RandomRouter
import akka.routing.RoundRobinRouter

object Example1 {
	def main(args: Array[String]): Unit = {}
	val _system = ActorSystem.create("balancing-dispatcher",ConfigFactory.load().getConfig("MyDispatcherExample"))

	val actor = _system.actorOf(Props[MsgEchoActor].withDispatcher("balancingDispatcher").withRouter(
						RoundRobinRouter(5)))

	0 to 25 foreach {
		i => actor ! i
	}
	_system.shutdown()
}

File: AkkaDispatcherExample/src/main/java/org/akka/essentials/dispatcher/example/BalancingDispatcher/Example1.java
package org.akka.essentials.dispatcher.example.BalancingDispatcher;

import org.akka.essentials.dispatcher.MsgEchoActor;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.routing.RoundRobinRouter;

import com.typesafe.config.ConfigFactory;

public class Example1 {
	/**
	 * @param args
	 */
	public static void main(String[] args) {
		ActorSystem _system = ActorSystem.create("balancing-dispatcher",
				ConfigFactory.load().getConfig("MyDispatcherExample"));
		
		ActorRef actor = _system.actorOf(new Props(MsgEchoActor.class)
				.withDispatcher("balancingDispatcher").withRouter(
						new RoundRobinRouter(5)));

		for (int i = 0; i < 25; i++) {
			actor.tell(i);
		}

		_system.shutdown();

	}
}
